File: src/main/java/com/example/socialnetworkingapp/model/education/Education.java
package com.example.socialnetworkingapp.model.education;

import com.sun.istack.NotNull;
import lombok.*;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name="education")
public class Education implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false, updatable = false, unique = true)
    private Long id ;

    @NotNull
    private String school;

    private String degree;

    private String field;

    private String startDate;

    private String endDate;

    private float grade;

    @Lob
    @Type(type = "org.hibernate.type.TextType")
    private String description;

    @NotNull
    private Boolean visible = true;
}

File: socialnetworkingapp-front/src/app/education/education.ts

export class Education {

  id: number;
  school: string;
  degree: string;
  field: string;
  startDate: string;
  endDate: string;
  grade: number;
  description: string;
  visible: boolean;

  constructor(id: number, school: string, degree: string, field: string, startDate: string, endDate: string, grade: number, description: string, visible: boolean) {
    this.id = id;
    this.school = school;
    this.degree = degree;
    this.field = field;
    this.startDate = startDate;
    this.endDate = endDate;
    this.grade = grade;
    this.description = description;
    this.visible = visible;
  }
}
